pyvrp/cpp/Solution.cpp

```diff
@@ -287,7 +279,18 @@ Duration Solution::Route::serviceDuration() const { return service_; }
 
 Duration Solution::Route::timeWarp() const { return timeWarp_; }
 
-Duration Solution::Route::waitDuration() const { return wait_; }
+Duration Solution::Route::waitDuration() const
+{
+    return duration_ - travel_ - service_;
+}
+
+Duration Solution::Route::travelDuration() const { return travel_; }
+
+Duration Solution::Route::startTime() const { return startTime_; }
+
+Duration Solution::Route::endTime() const { return startTime_ + duration_; }
```

Comment:
I think this is only correct if `timeWarp = 0`, it should be `endTime = startTime + duration - timeWarp` if we want this to be consistent with the definition of `endTime` in `TimeWindowSegment` (defined implicitly in `merge` as the start time for the travel).

See this code in `TimeWindowSegment::merge`:
```
auto const travelDuration = durationMatrix(idxLast, other.idxFirst);
auto const diff = duration - timeWarp + travelDuration;
auto const diffWait = std::max<Duration>(other.twEarly - diff - twLate, 0);
auto const diffTw = std::max<Duration>(twEarly + diff - other.twLate, 0);
```
The arrival time at the first client of the other segment is computed as `twEarly + diff`, which is `twEarly + duration - timeWarp + travelDuration` suggesting that `twEarly + duration - timeWarp` is the endTime corresponding to the starttime `twEarly`

We could consider a test for this (I can propose one but it may take a few days...).

Reply:
I've added this in, along with a test!